extension/observer/endpointswatcher.go

```diff
@@ -31,7 +31,7 @@ type EndpointsWatcher struct {
 	logger            *zap.Logger
 }
 
-func NewEndpointsWatcher(endpointsLister EndpointsLister, refreshInterval time.Duration, logger *zap.Logger) *EndpointsWatcher {
+var NewEndpointsWatcher = func(endpointsLister EndpointsLister, refreshInterval time.Duration, logger *zap.Logger) *EndpointsWatcher {
```

Comment:
That's cheating :) there is no functional difference here. Since this is acting as a library, maybe we can discuss moving this to internal.

Reply:
Thanks @atoulme! this sounds more reasonable since `NewEndpointsWatcher` is being used by multiple observers, reopen issue [29083](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29083) here.

Reply:
Should i close or draft this PR since we maybe need to move this to internal?